tofu/geom/_GG02.pyx

```diff
@@ -2957,302 +2876,13 @@ def integrate1d(y, double dx, t=None, str method='sum'):
-                    sig[:,ii] = np.sum(ff(pts,t=t,**fkwdargs),axis=axm)*dl
-
-            elif method=='simps':
-                for ii in range(0,ND):
-                    dl0 = DLs[0,ii]
-                    # Compute the number of intervals to satisfy the resolution
-                    N = <long>(Cceil((DLs[1,ii]-dl0)/dL[ii]))
-                    N = N if N%2==0 else N+1
-                    dl = (DLs[1,ii]-dl0)/<double>N
-                    D0, D1, D2 = Ds[0,ii], Ds[1,ii], Ds[2,ii]
-                    u0, u1, u2 = us[0,ii], us[1,ii], us[2,ii]
-                    pts = np.empty((3,N+1),dtype=float)
-                    for jj in range(0,N+1):
-                        kkk = dl0 + (<double>jj)*dl
-                        pts[0,jj] = D0 + kkk*u0
-                        pts[1,jj] = D1 + kkk*u1
-                        pts[2,jj] = D2 + kkk*u2
-                    if ani:
-                        fkwdargs['Vect'] = (-u0,-u1,-u2)
-                    sig[:,ii] = scpintg.simps(ff(pts,t=t,**fkwdargs),
-                                              x=None,dx=dl,axis=axm)
-
-            else:
-                for ii in range(0,ND):
-                    dl0 = DLs[0,ii]
-                    # Compute the number of intervals to satisfy the resolution
-                    N = <long>(Cceil((DLs[1,ii]-dl0)/dL[ii]))
-                    N = 2**(<long>(Cceil(Clog2(<double>N))))
-                    dl = (DLs[1,ii]-dl0)/<double>N
-                    D0, D1, D2 = Ds[0,ii], Ds[1,ii], Ds[2,ii]
-                    u0, u1, u2 = us[0,ii], us[1,ii], us[2,ii]
-                    pts = np.empty((3,N+1),dtype=float)
-                    for jj in range(0,N+1):
-                        kkk = dl0 + (<double>jj)*dl
-                        pts[0,jj] = D0 + kkk*u0
-                        pts[1,jj] = D1 + kkk*u1
-                        pts[2,jj] = D2 + kkk*u2
-                    if ani:
-                        fkwdargs['Vect'] = (-u0,-u1,-u2)
-                    sig[:,ii] = scpintg.romb(ff(pts,t=t,**fkwdargs),
-                                             dx=dl,axis=axm,show=False)
-
-    if nt==1:
-        return sig.ravel()
-    else:
-        return sig
-
-
-
-
-def LOS_calc_signal2(func, double[:,::1] Ds, double[:,::1] us, dL,
-                     double[:,::1] DLs, str dmethod='abs',
-                     str method='sum', bint ani=False,
-                     t=None, fkwdargs={}, str minimize='calls',
-                     bint Test=True, int num_threads=16):
+def LOS_calc_signal(func, double[:,::1] Ds, double[:,::1] us, dL,
+                    double[:,::1] DLs, str dmethod='abs',
+                    str method='sum', bint ani=False,
+                    t=None, fkwdargs={}, str minimize='calls',
+                    bint Test=True, int num_threads=16):
     """ Compute the synthetic signal, minimizing either function calls or memory
+    TODO: since we are working in cython... the least is to give "func" 's signature
```

Comment:
As said in the comments: the signature of the function `func` should be given

Reply:
func(pts, t=None, vect=None) => data 
pts : ndarray (3, n)
vect : ndarray(3, n)
t = ndarray(m)
data : ndarray(n) if t is None, else ndarray(m,n)